pages/api/plants/[id]/index.js

```diff
@@ -10,55 +10,85 @@ export default async function handler(request, response) {
       req: request,
       secret: process.env.NEXTAUTH_SECRET,
     });
-    if (request.method === "GET") {
-      const plant = await Plant.findById(id);
-      if (!plant) {
-        return response
-          .status(404)
-          .json({ success: false, message: "Plant not found" });
-      }
-      //Private plants are only accessible by admins!
-      if (!plant.isPublic && token?.role !== "admin") {
+    switch (request.method) {
+      case "GET": {
+        const plant = await Plant.findById(id);
+        if (!plant) {
+          return response
+            .status(404)
+            .json({ success: false, message: "Plant not found" });
+        }
+        //Private plants are only accessible by admins!
+        if (!plant.isPublic && (!token || token?.role !== "admin")) {
           return response.status(403).json({ error: "Forbidden" });
+        }
+        return response.status(200).json(plant);
       }
-      return response.status(200).json(plant);
-    }
-
-    //every reqest method except "GET" is protected
-
-    if (!token) {
-      return response.status(401).json({ error: "Not authenticated" });
-    }
-    if (token.role !== "admin") {
-      return response.status(403).json({ error: "Forbidden" });
-    }
+      //From here on only admins can use the api methods
+      default: {
+        if (!token) {
+          return response.status(401).json({ error: "Not authenticated" });
+        }
+        if (token.role !== "admin") {
+          return response.status(403).json({ error: "Forbidden" });
+        }
 
-    if (request.method === "PUT") {
-      const editedPlant = { ...request.body, isPublic: true };
-      const plant = await Plant.findByIdAndUpdate(id, editedPlant, {
-        new: true,
-        runValidators: true,
-      });
-      if (!plant) {
-        return response
-          .status(404)
-          .json({ success: false, message: "Plant not found" });
-      }
-      return response.status(200).json({ success: true, data: plant });
-    } else if (request.method === "DELETE") {
-      const deleted = await Plant.findByIdAndDelete(id);
-      if (!deleted) {
-        return response
-          .status(404)
-          .json({ success: false, message: "Plant not found" });
+        switch (request.method) {
+          //Use PUT for Edit Plant
+          case "PUT": {
```

Comment:
I don't recommend nested switch/case block. I recommend to differentiate inside one case block to handle admin vs user behavior.